pandas/_libs/tslib.pyx

```diff
@@ -744,15 +744,7 @@ cdef _array_to_datetime_object(
 
         cnp.PyArray_MultiIter_NEXT(mi)
 
-    warnings.warn(
-        "In a future version of pandas, parsing datetimes with mixed time "
-        "zones will raise an error unless `utc=True`. "
-        "Please specify `utc=True` to opt in to the new behaviour "
-        "and silence this warning. To create a `Series` with mixed offsets and "
-        "`object` dtype, please use `apply` and `datetime.datetime.strptime`",
-        FutureWarning,
-        stacklevel=find_stack_level(),
-    )
+    raise ValueError("cannot parse datetimes with mixed time zones unless `utc=True`")
```

Comment:
Since this method is only used once when there are mixed timezones, this entire function can be removed

Reply:
Thank you for reviewing this PR. I removed the function `_array_to_datetime_object`.